lib/health_cards/health_card.rb

```diff
@@ -11,43 +17,87 @@ class << self
       # @return [HealthCards::HealthCard]
       def from_jws(jws, public_key: nil, key: nil)
         jws = JWS.from_jws(jws, public_key: public_key, key: key)
-        vc = VerifiableCredential.decompress_credential(jws.payload)
-        HealthCard.new(verifiable_credential: vc, jws: jws)
+        from_payload(jws.payload)
+      end
+
+      def from_payload(payload)
+        inf = Zlib::Inflate.new(-Zlib::MAX_WBITS).inflate(payload)
+        json = JSON.parse(inf)
+
+        bundle_hash = json.dig('vc', 'credentialSubject', 'fhirBundle')
+
+        raise InvalidCredentialError unless bundle_hash
+
+        bundle = FHIR::Bundle.new(bundle_hash)
+        new(issuer: json['iss'], bundle: bundle)
+      end
+
+      def allow(klass, attributes)
+        @allowable ||= {}
```

Comment:
You don't need to do this since you have the method below.

Reply:
will fix